src/contrib/cluster/Akka.Cluster.Tools.Tests.MultiNode/ClusterClient/ClusterClientDiscoverySpec.cs

```diff
@@ -0,0 +1,297 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="ClusterClientHandoverSpec.cs" company="Akka.NET Project">
+//     Copyright (C) 2009-2023 Lightbend Inc. <http://www.lightbend.com>
+//     Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Linq;
+using Akka.Actor;
+using Akka.Cluster.TestKit;
+using Akka.Cluster.Tools.Client;
+using Akka.Cluster.Tools.PublishSubscribe;
+using Akka.Configuration;
+using Akka.Discovery;
+using Akka.Discovery.Config;
+using Akka.MultiNode.TestAdapter;
+using Akka.Remote.TestKit;
+using Akka.TestKit.TestActors;
+using FluentAssertions;
+
+namespace Akka.Cluster.Tools.Tests.MultiNode.Client
+{
+    public sealed class ClusterClientDiscoverySpecConfig : MultiNodeConfig
+    {
+        public RoleName Client { get; }
+        public RoleName First { get; }
+        public RoleName Second { get; }
+        public RoleName Third { get; }
+
+        public ClusterClientDiscoverySpecConfig()
+        {
+            Client = Role("client");
+            First = Role("first");
+            Second = Role("second");
+            Third = Role("third");
+
+            CommonConfig = ConfigurationFactory.ParseString("""
+akka.loglevel = INFO
+
+akka.remote.dot-netty.tcp.hostname = localhost 
+akka.actor.provider = cluster
+akka.remote.log-remote-lifecycle-events = off
+akka.cluster.client {
+  heartbeat-interval = 1d
+  acceptable-heartbeat-pause = 1d
+  reconnect-timeout = 3s
+  refresh-contacts-interval = 1d
+}
+akka.test.filter-leeway = 10s
+""")
+                .WithFallback(ClusterClientReceptionist.DefaultConfig())
+                .WithFallback(DistributedPubSub.DefaultConfig())
+                .WithFallback(MultiNodeClusterSpec.ClusterConfig());
+            
+            NodeConfig(new[]{ Client }, 
+                new []{
+                    ConfigurationFactory.ParseString("""
+akka {
+  cluster.client {
+    heartbeat-interval = 1s
+    acceptable-heartbeat-pause = 2s
+    use-initial-contacts-discovery = true
+    reconnect-timeout = 4s
+    verbose-logging = true
+    discovery
+    {
+      service-name = test-cluster
+      discovery-timeout = 10s
+    }
+  }
+  
+
+  discovery
+  {
+    method = config
+    config.services.test-cluster.endpoints = []
```

Comment:
I assume this gets populated later using the new internal methods you added for dynamically updating config-based discovery